index.d.ts

```diff
@@ -14,6 +14,97 @@ declare const listenerAdded: unique symbol;
 declare const listenerRemoved: unique symbol;
 type OmnipresentEventData = {[listenerAdded]: Emittery.ListenerChangedData; [listenerRemoved]: Emittery.ListenerChangedData};
 
+/**
+Emittery can collect and log debug information.
+
+To enable this feature set the DEBUG environment variable to 'emittery' or '*'. Additionally you can set the static `isDebug` variable to true on the Emittery class, or `myEmitter.debug.enabled` on an instance of it for debugging a single instance.
+
+See API for more information on how debugging works
+*/
+type DebugLogger = (type: string, debugName: string, eventName?: EventName, eventData?: Record<string, any>) => void;
+
+/**
+Configure debug options of an instance
+*/
+interface DebugOptions {
+	/**
+	Define a name for the instance of Emittery to use when outputting debug data.
+
+	@default Empty string
+	@example
+	```
+	const Emittery = require('emittery');
+	Emittery.isDebug = true;
+	const emitter = new Emittery({debug: {name: 'myEmitter'}});
+	emitter.on('test', data => { // do something });
+	//=> [16:43:20.417][emittery:subscribe][myEmitter] Event Name: test
+	//	data: undefined
```

Comment:
This is too cramped. Use more empty lines to make it more readable. The example should also include the part that emits the event. Generally, code examples should be runnable on their own.

Applies to all code examples.